core/primitives/src/utils.rs

```diff
@@ -506,21 +504,23 @@ pub fn check_access_key(
     access_key: Option<AccessKey>,
     account_id: &AccountId,
     public_key: &PublicKey,
+    protocol_version: ProtocolVersion,
 ) -> Result<AccessKey, InvalidAccessKeyError> {
     match access_key {
         Some(access_key) => Ok(access_key),
         None => {
-            #[cfg(feature = "protocol_feature_eth_implicit")]
-            if account_id.get_account_type() == AccountType::EthImplicitAccount {
-                if derive_account_id_from_public_key(public_key) == *account_id {
-                    // TODO What about increasing storage usage for that account, because we added access key?
-                    return Ok(AccessKey::full_access());
-                } else {
-                    // Provided public key is not the one from which the signer address was derived.
-                    return Err(InvalidAccessKeyError::InvalidPkForEthAddress {
-                        account_id: account_id.clone(),
-                        public_key: public_key.clone(),
-                    });
+            if checked_feature!("stable", EthImplicit, protocol_version) {
```

Comment:
nit: recommend avoiding nested if by reversing the statement for better readability